test/dom/as.mocha.js

```diff
@@ -1,45 +1,59 @@
-var expect = require('chai').expect;
-var derby = require('./util').derby;
+const expect = require('chai').expect;
+const domTestRunner = require('../../src/test-utils/domTestRunner');
 
 describe('as', function() {
+  const runner = domTestRunner.install({
+    jsdomOptions: {
+      // solution for `SecurityError: localStorage is not available for opaque origins`
+      // Racer interfaces with localStorage and the `as-object` tests use `page.model`
+      // methods causing the SecurityError if `url` is not set. Does not appear to impact
+      // `as-array` tests even though they also use `page.model` methods so 🤷
+      url: 'http://localhost/'
+    }
+  });
+
   it('HTML element `as` property', function() {
-    var app = derby.createApp();
+    const { app } = runner.createHarness();
     app.views.register('Body', '<div as="nested[0]"></div>');
-    var page = app.createPage();
-    var fragment = page.getFragment('Body');
+    const page = app.createPage();
+    const fragment = page.getFragment('Body');
     expect(page.nested[0]).html('<div></div>');
     expect(fragment).html('<div></div>');
   });
 
   it('Component `as` property', function() {
-    var app = derby.createApp();
+    const { app } = runner.createHarness();
     app.views.register('Body', '<view is="item" as="nested[0]"></view>');
     app.views.register('item', '<div></div>')
     function Item() {};
     app.component('item', Item);
-    var page = app.createPage();
-    var fragment = page.getFragment('Body');
+    const page = app.createPage();
+    const fragment = page.getFragment('Body');
     expect(page.nested[0]).instanceof(Item);
     expect(page.nested[0].markerNode.nextSibling).html('<div></div>');
     expect(fragment).html('<div></div>');
   });
 
-  it('HTML element `as-object` property', function() {
-    var app = derby.createApp();
+  async function nextTick() {
```

Comment:
Let's add a comment explaining why this is needed in these tests:
https://github.com/derbyjs/derby/blob/59db1ad8a1962b58f493c73f040046135063b4ec/src/templates/templates.ts#L2342